src/codecatalyst/activation.ts

```diff
@@ -96,3 +98,29 @@ export async function activate(ctx: ExtContext): Promise<void> {
         authProvider.removeSavedConnection()
     })
 }
+
+async function showReadmeFileOnFirstLoad(workspaceState: vscode.ExtensionContext['workspaceState']): Promise<void> {
+    // Check dev env state to see if this is the first time the user has connected to a dev env
+    const isFirstLoad = workspaceState.get('aws.codecatalyst.devEnv.isFirstLoad', true)
```

Comment:
Not a blocker: thinking about edge cases: does workspaceState store in the project-local .vscode folder, rather than the user-global (but remote) state? Ideally it would be stored in the user-global (but remote) globalstate, since this is an environment-global state. And it's possible that the workspace state or "main project" changes, e.g. on branch-switch or other manual actions.

Maybe a future idea, could we avoid this state if we checked that the env was created in the last ~1 hour? I guess that's too flaky though.

Reply:
`workspaceState` _might_ live in the home directory of the remote. If so, it shouldn't be used in a dev environment. One way to check is to nuke `$HOME` in the remote and then restart the environment.